oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/MapAggregator.java

```diff
@@ -220,17 +220,15 @@ private MapAggregator(MapAggregator<U, ?> obj, MapReducer<IndexValuePair<U, X>>
       );
     } else {
       MapAggregator<OSHDBCombinedIndex<U, V>, ? extends OSHDBMapReducible> ret;
-      if (this.mapReducer.forClass.equals(OSMContribution.class)) {
-        ret = this.flatMap(x -> gs.splitOSMContribution((OSMContribution) x).entrySet())
+      if (mapReducer.isContributionViewQuery()) {
+        ret = this.flatMap(x -> gs.splitOSMContribution((OSMContributionImpl) x).entrySet())
```

Comment:
Use the interface instead of the implementing class for the cast, except there is a reason why not. I think it was a left over from search and replace, happens in the MapReducer as well?

```suggestion
        ret = this.flatMap(x -> gs.splitOSMContribution((OSMContribution) x).entrySet())
```